### PR TITLE
SCB-1690 omega-transport-resttemplate does not use RestTemplateBuilder

### PR DESCRIPTION
**Pull request by chanjarster**
Issue details: https://issues.apache.org/jira/browse/SCB-1690

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java [hunk 1]
```diff
@@ -18,25 +18,22 @@
...
 
   @Bean(name = "omegaRestTemplate")
-  public RestTemplate omegaRestTemplate(@Autowired(required=false) OmegaContext context) {
-    RestTemplate template = new RestTemplate();
-    List<ClientHttpRequestInterceptor> interceptors = template.getInterceptors();
-    interceptors.add(new TransactionClientHttpRequestInterceptor(context));
-    template.setInterceptors(interceptors);
-    return template;
+  public RestTemplate omegaRestTemplate(@Autowired(required = false) OmegaContext context,
+      RestTemplateBuilder restTemplateBuilder) {
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java]: RestTemplateBuilder is based in the spring-boot, why do we need to introduce the dependency of spring-boot-autoconfigure.

Comment (chanjarster) [reply omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java]: Yeah, we don't need it.

File: omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java [hunk 2]
```diff
@@ -18,25 +18,22 @@
...
-import org.springframework.http.client.ClientHttpRequestInterceptor;
 import org.springframework.web.client.RestTemplate;
 
-import org.apache.servicecomb.pack.omega.context.OmegaContext;
-
 @Configuration
 public class RestTemplateConfig {
 
   @Bean(name = "omegaRestTemplate")
-  public RestTemplate omegaRestTemplate(@Autowired(required=false) OmegaContext context) {
```
Comment (coolbeevip) [inline omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java]: We should keep this method and add @ConditionalOnMissingBean

Comment (chanjarster) [reply omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java]: You mean `@ConditionalOnMissingBean(RestTemplateBuilder.class)` ?

Comment (WillemJiang) [reply omega/omega-transport/omega-transport-resttemplate/src/main/java/org/apache/servicecomb/pack/omega/transport/resttemplate/RestTemplateConfig.java]: RestTemplateBuilder is in the default dependency of spring boot, I think it is OK we keep the code here.

**Pull request merged**